core-bundle/src/Command/MigrateCommand.php

```diff
@@ -537,4 +546,40 @@ private function compileTableWarnings(Table $table): array
 
         return $warnings;
     }
+
+    private function validateDatabaseVersion(bool $asJson): bool
+    {
+        // TODO: Find a replacement for getWrappedConnection() once doctrine/dbal 4.0 is released
+        $driverConnection = $this->connection->getWrappedConnection();
+        $currentPlatform = \get_class($this->connection->getDatabasePlatform());
+        $driver = $this->connection->getDriver();
+
+        if (
+            !$driverConnection instanceof ServerInfoAwareConnection
+            || !$driver instanceof VersionAwarePlatformDriver
+        ) {
+            return true;
+        }
+
+        $version = $driverConnection->getServerVersion();
+        $correctPlatform = \get_class($driver->createDatabasePlatformForVersion($version));
+
+        if ($correctPlatform === $currentPlatform) {
+            return true;
+        }
+
+        $message = sprintf('Wrong database version configured, please set it to "%s"', $version);
+
+        if ($currentVersion = $this->connection->getParams()['serverVersion'] ?? null) {
+            $message .= sprintf(', currently set to "%s"', $currentVersion);
+        }
+
+        if ($asJson) {
+            $this->writeNdjson('database-error', ['message' => $message]);
```

Comment:
The correct name needs to be resolved in https://github.com/contao/contao/pull/5065#discussion_r940079950 first

Reply:
```suggestion
            $this->writeNdjson('problem', ['message' => $message]);
```